server/routerlicious/packages/services-utils/src/runner.ts

```diff
@@ -73,7 +65,17 @@ export async function run<T extends IResources>(
     const runner = await runnerFactory.create(resources);
 
     // Start the runner and then listen for the message to stop it
-    const runningP = runner.start(logger);
+    const runningP = runner
+        .start(logger)
+        .catch(async (error) => {
+            await runner
+                    .stop()
+                    .catch((innerError) => {
+                        logger?.error(`Could not stop runner due to error: ${innerError}`);
+                    });
+            return Promise.reject(error);
+        });
+
```

Comment:
Can we consolidate the logic in one place? Rather than rejecting the promise, we can log the error and call process.exit()?

Reply:
That is a good suggestion, but I'm wondering if it's possible for the `run()` method to fail outside of this `runner.start().catch()` scope. For example, if we move logging/`process.exit()` logic from `runService()`, and this part fails for some reason:

```
process.on("SIGTERM", () => {
        // eslint-disable-next-line @typescript-eslint/no-floating-promises
        runner.stop();
    });
```

then `runService()` would not be able to handle that kind of failure anymore. What do you think?

Reply:
Ah I see. I thought there are in same block. But the last check in actually outside. Makes sense